[Bryan Yang] iP

TheSoggy opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/admin/ip-w4.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is ****FREE****!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
-----------------------------------------

If you Java programmer, you can <mark>use</mark> it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

TheSoggy committed:
Add functionality to echo user input and exit upon specific input

TheSoggy committed:
Add functionality to store user inputs and display them back to the user when requested

TheSoggy committed:
Add functionality to mark tasks as done/not done

TheSoggy committed:
Amend javadocs

TheSoggy committed:
Add support for tracking three types of tasks

TheSoggy committed:
Edit javadocs

TheSoggy committed:
Add automated text UI testing

TheSoggy committed:
Handle errors with DukeException class

TheSoggy committed:
Add support for deleting tasks from the list.

TheSoggy committed:
Use enum to represent CommandType

Devanshshah1309 commented on src/main/java/IOParser.java:
```diff
@@ -0,0 +1,22 @@
+
+import java.util.Arrays;
+
+public class IOParser {
```
Heyy, any reason the `IOParser` doesn't parse the user's input? I think parsing of user input is being done in `Duke` as of now, right?

Devanshshah1309 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,10 @@
+public class Main {
```
Any reason for not using the `main` method of `Duke` class?

Devanshshah1309 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,100 @@
+        return String.format("[%c][%s] %s", symbol, getStatusIcon(), description);
+    }
+
+    private static class TodoTask extends Task {
```
Would it be better to have these as separate files rather than static nested classes?

Devanshshah1309 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,100 @@
+        }
+    }
+
+    private static class DeadlineTask extends Task {
```
Similarly for this too

Devanshshah1309 reviewed (commented):
Great work :) Clean and understandable code 👍

TheSoggy committed:
Refactor Task class into subclasses and add method to show their format in data file

TheSoggy committed:
Add functionality to save tasks into hard disk

TheSoggy committed:
Add functionality to parse DateTime

TheSoggy committed:
Merge branch 'branch-Level-8'

TheSoggy committed:
Merge branch 'branch-Level-7' into HEAD

TheSoggy committed:
Merge branch 'branch-Level-8' into HEAD

TheSoggy committed:
Merge branch 'master' of https://github.com/TheSoggy/ip

TheSoggy committed:
Divide classes into packages

TheSoggy committed:
Add Gradle support

TheSoggy committed:
Refactor code to extract out Storage class

TheSoggy committed:
Exclude data folder from revision control

TheSoggy committed:
Add functionality to print number of tasks after loading file

TheSoggy committed:
Refactor command functionality into Command class

TheSoggy committed:
Remove wildcard import

TYKCodes commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,57 @@
+
+public class Storage {
+
+    private static final String taskListTxt = "duke.txt";
```
Perhaps the name of your constant variable should be capitalised? Maybe something like:
```suggestion
    private static final String TASKLIST_TXT = "duke.txt";
```

TYKCodes commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import duke.command.AddTaskCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.CommandType;
+import duke.command.DeleteTaskCommand;
+import duke.command.EmptyCommand;
+import duke.command.ListCommand;
+import duke.command.SetDoneCommand;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
```
It's great that you listed all of the classes that you are importing instead of using a wildcard

TYKCodes reviewed (commented):
Overall, your code looks clean and understandable to read. Some of your classes and its methods are missing some documentation, but otherwise the other descriptions are well done and descriptive enough to understand

TheSoggy committed:
Rename boolean variable to ensure coding standard

TheSoggy committed:
Add JUnit Tests for Parser and TaskList

TheSoggy committed:
Update gradle to accept UTF-8 encoding

TheSoggy committed:
Add functionality to find tasks by searching for a keyword

TheSoggy committed:
Remove unused imports

TheSoggy committed:
Merge branch 'branch-A-JavaDoc'

TheSoggy committed:
Refactor code to adhere to coding standards

TheSoggy committed:
Merge branch 'branch-A-CodingStandard'

TheSoggy committed:
Add functionality to find tasks by searching for a keyword

TheSoggy committed:
Merge branch 'branch-Level-9'

TheSoggy committed:
Add CheckStyle and modify code accordingly

TheSoggy committed:
Merge branch 'branch-A-CheckStyle'

TheSoggy committed:
Add private constructor to Parser

TheSoggy committed:
Add JavaFx to build.gradle

TheSoggy committed:
Refactor storage to store Java objects in duke.dat

TheSoggy committed:
Implement GUI support

TheSoggy committed:
Merge branch 'branch-Level-10'

TheSoggy committed:
Fix bug for Duke to create new TaskList if file storage does not exist

TheSoggy committed:
Personalise UI

Current UI is bland.

A more personalised UI will differentiate the app from others.

Let's,
- Change background color of Scene and DialogBox
- Change icon shape to be circle
- Change border color
- Change text color to suit background color
- Change width of scene

TheSoggy committed:
Remove text-UI based tests

Text-UI based tests have become irrelevant ever since the introduction
of the GUI. This patch removes them.

TheSoggy committed:
Add assertions

TheSoggy committed:
Refactor code to improve code quality

TheSoggy committed:
Refactor code to use Java streams

TheSoggy committed:
Merge pull request #3 from TheSoggy/branch-A-CodeQuality

Refactor code to improve code quality

TheSoggy committed:
Merge pull request #4 from TheSoggy/branch-A-Streams

Refactor code to use Java streams

TheSoggy committed:
Merge remote-tracking branch 'origin' into branch-A-Assertions

TheSoggy committed:
Merge pull request #2 from TheSoggy/branch-A-Assertions

Add assertions

TheSoggy committed:
Set up CI

TheSoggy committed:
Merge branch 'master' of https://github.com/TheSoggy/ip

TheSoggy committed:
Update the execution permission for gradlew

TheSoggy committed:
Fix code style issues

TheSoggy committed:
Allow GUI to be resizeable

TheSoggy committed:
Refactor find command to use Java streams

TheSoggy committed:
Remove toData method from Task given that Task is no longer stored as a string

TheSoggy committed:
Edit find/list command to handle empty TaskList

TheSoggy committed:
Edit bye command to make output more interactive

Show a countdown timer (3s) when Duke is closing

TheSoggy committed:
Personalise UI further

TheSoggy committed:
More UI personalisation

TheSoggy committed:
Refactor Parser code

TheSoggy committed:
Add sort functionality to TaskList based on description or time

TheSoggy committed:
Fix code style issues

TheSoggy committed:
Rename classes and fix bug in unmark command

TheSoggy committed:
Add user guide

TheSoggy committed:
Remove unused file

TheSoggy committed:
Fix bug in unmark command

TheSoggy committed:
Fix code style issues

TheSoggy committed:
Update README.md

TheSoggy committed:
Update README.md

TheSoggy committed:
Add Ui.png

TheSoggy committed:
Update Javafx plugin version
